MaxText/tests/hf_checkpoint_conversion_check.py

```diff
@@ -70,46 +71,181 @@ def get_logits(inputs, model, golden_model):
   return logits, golden_logits
 
 
-def main(argv: Sequence[str]) -> None:
-  # Parse arguments from argv
-  # Default values
-  parsed_args = {"golden_model_id": "google/gemma-2-2b-it", "hf_checkpoint_path": os.path.expanduser("~/.hf_output/")}
-  for arg in argv[1:]:
-    if "=" in arg:
-      key, value = arg.split("=", 1)
-      if key in parsed_args:
-        parsed_args[key] = value
-      else:
-        print(f"Warning: Unknown argument '{key}' found in argv. Ignoring.")
-
-  golden_model = AutoModelForCausalLM.from_pretrained(parsed_args["golden_model_id"], torch_dtype=torch.float32)
-
-  tokenizer = AutoTokenizer.from_pretrained(parsed_args["hf_checkpoint_path"])
-  model = AutoModelForCausalLM.from_pretrained(parsed_args["hf_checkpoint_path"], torch_dtype=torch.float32)
-
-  # TODO: (@yixuannwang) use 3 prompts to verify
-  input_text = "I love to"
-  inputs = tokenizer(input_text, return_tensors="pt")
-  # --- Generate Output ---
-  with torch.no_grad():
-    outputs = model.generate(**inputs, max_new_tokens=8)
-  # --- Decode and Print ---
-  print(tokenizer.decode(outputs[0], skip_special_tokens=True))
-
-  # Check weights match
-  print("########### check weights match ############### ")
-  check_weights_match(model, golden_model)
+def get_top_k_tokens_scores(logits_tensor, tokenizer_instance, k=10, description=""):
+  """Get the top-k tokens and their scores from a given logits tensor."""
+  print(f"\n--- {description} top {k} tokens ---")
+  collected_tokens = []
+  tokens = []
+  # Ensure logits_tensor is on CPU for operations like topk and item()
+  logits_tensor = logits_tensor.cpu()
+  topk_results = torch.topk(logits_tensor[0, -1], k=k)
+  for i in range(k):
+    tok_id = topk_results.indices[i].item()
+    score = topk_results.values[i].item()
+    tok = tokenizer_instance.decode(tok_id)
+    collected_tokens.append({"id": int(tok_id), "token": tok.strip(), "score": float(score)})
+    tokens.append({"id": int(tok_id), "token": tok.strip(), "score": float(score)})
+
+  # Prepare data for tabulate: a list of lists
+  table_data = [[d["id"], d["token"], d["score"]] for d in collected_tokens]
+  print(tabulate(table_data, headers=["Token ID", "Token", "Score"], tablefmt="orgtbl"))
```

Comment:
Nit: I probably forgot to mention in previous PR. Could we use `max_logging.log` to align in the repo?